### DISCUSSION TITLE
cocotb coherence

### DISCUSSION THREAD

**Discussion by toddstrader** [cocotb/cocotb-bus#64] [General]
Is there an expectation for HEAD-to-HEAD correctness between cocotb and cocotb-bus?  Is coherence expected to be best effort?  Will the packages be released in tandem so at least releases are known to work together?  Also, I'm not sure if this is possible with GitHub CI, but should this repo know when it is broken wrt cocotb's HEAD?

Also also, what's the prognosis for this repo?  I see it hasn't had a commit in over a year.  Dare I ask, is cocotb-bus dead?

I recently posted #62 when `fork` was removed from cocotb.  And now I'm seeing that `_AssignmentResult` has been removed with is breaking `bus.py`.  I haven't dug further since the last round of pruning in cocotb, so I'm not sure what else is busted.  I can keep fixing things up, but just want to make sure I'm headed in a meaningful direction.

@ktbarrett thoughts?

**Comment by ktbarrett** [comment]
We need a maintainer willing to work on this repo. And I don't think that cocotb-bus and cocotb should be able to be installed together right now, the setup.py in this repo lists `<2.0`.

EDIT: Ah, 2.0.dev0 *is* less than 2.0.

**Comment by jahagirdar** [comment]
@ktbarrett @toddstrader ,
I too am interested in the future of this repo. To make cocotb VIP's reusable, it is better if they follow a similar pattern, By providing the basic building blocks this repo helps in that direction.
Looking at the VIP's I use, almost all of them have spun their own bus system(AXI,SPI) or require complicated invocations to do basic stuff(AHB, Wishbone)
If we can have a discussion on the future direction for this repo, I volunteer to spend some time on this.

**Comment by gbowers4** [comment]
I agree @jahagirdar  - I think providing a basic set of tools to implement VIPs is a good direction, happy to put in some work to get this functionality alongside the cocotb project.

**Comment by jahagirdar** [comment]
@eric-wieser @cmarqu @imphil ,
Is it possible to enable github discussions on this repo. This will facilitate a structured discussion on the future development.

**Comment by cmarqu** [comment]
Good idea, I just enabled them.

**Comment by davekeeshan** [comment]
What is the status of this repo? Are there any regular maintainers, I think a release if due to keep compatibility with cocotb, the last release for cocotb-bus was Jan 2022.

Are you looking for maintainers, I may be able to look at some issues, I have been using cocotb-bus a lot recently but mainly have been using my own forked version due to the compatibility issues.
